lib/basedriver/commands/event.js

```diff
@@ -14,13 +16,22 @@ commands.logCustomEvent = function (vendor, event) {
 /**
  * Get the event log
  *
- * @param {string} type - currently unused but a placeholder for a future
- * feature where filtering of log events is allowed
+ * @param {?string|Array<string>} type - the event type to filter with.
+ * It returns all events if the type is not provided or empty string/array.
  * @returns {object} - the event history log object
  */
-commands.getLogEvents = function (/*type*/) {
-  // type is currently unused but may be implemented for filtering later on
-  return this._eventHistory;
+commands.getLogEvents = function (type) {
+  if (_.isEmpty(type)) {
+    return this._eventHistory;
+  }
+
+  const events = {};
```

Comment:
use _.reduce here. See https://github.com/appium/appium-base-driver/blob/5b12eebcf011cf7676a2987b646a6394ddafb639/lib/basedriver/commands/log.js#L26 for an example

Reply:
Ah, nice. I checked only `filter`, but it was not available for Object...